### PR TITLE
Improve perf of noise texture gen in lens and ssao

### PR DESCRIPTION
**Pull request by bghgary**
Use raw texture with a buffer of pixels instead of dynamic texture to render one pixel at a time using canvas.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts [hunk 1]
```diff
@@ -2,18 +2,20 @@
...
 import type { Effect } from "../../../Materials/effect";
 import { Texture } from "../../../Materials/Textures/texture";
-import { DynamicTexture } from "../../../Materials/Textures/dynamicTexture";
 import type { RenderTargetTexture } from "../../../Materials/Textures/renderTargetTexture";
 import { PostProcess } from "../../../PostProcesses/postProcess";
 import { PostProcessRenderPipeline } from "../../../PostProcesses/RenderPipeline/postProcessRenderPipeline";
 import { PostProcessRenderEffect } from "../../../PostProcesses/RenderPipeline/postProcessRenderEffect";
 import type { Scene } from "../../../scene";
+import { RawTexture } from "core/Materials/Textures/rawTexture";
+import { Constants } from "core/Engines/constants";
```
Comment (sebavan) [inline packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts:11]: core -> ../../../

File: packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts [hunk 2]
```diff
@@ -2,18 +2,20 @@
...
 
 import "../../../PostProcesses/RenderPipeline/postProcessRenderPipelineManagerSceneComponent";
 
 import "../../../Shaders/chromaticAberration.fragment";
 import "../../../Shaders/lensHighlights.fragment";
 import "../../../Shaders/depthOfField.fragment";
+import { Scalar } from "core/Maths/math.scalar";
```
Comment (sebavan) [inline packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts:18]: core -> ../../../

**Review by sebavan — APPROVED** [review]
THANKS A TON !!! :-) 

Aside of core -> core -> ../../../ a bit all over it looks all good.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14300/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14300/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14300/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14300/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14300/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14300/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14300/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14300/merge#BCU1XR#0

File: packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssaoRenderingPipeline.ts [hunk 3]
```diff
@@ -340,29 +341,17 @@ export class SSAORenderingPipeline extends PostProcessRenderPipeline {
...
-            }
+        const data = new Uint8Array(size * size * 3);
+        for (let index = 0; index < data.length; ) {
+            data[index++] = Math.floor(Math.max(0.0, Scalar.RandomRange(-1.0, 1.0)) * 255);
+            data[index++] = Math.floor(Math.max(0.0, Scalar.RandomRange(-1.0, 1.0)) * 255);
+            data[index++] = Math.floor(Math.max(0.0, Scalar.RandomRange(-1.0, 1.0)) * 255);
         }
 
-        this._randomTexture.update(false);
+        const texture = RawTexture.CreateRGBTexture(data, size, size, this._scene, false, false, Constants.TEXTURE_BILINEAR_SAMPLINGMODE);
```
Comment (bghgary) [inline packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssaoRenderingPipeline.ts:351]: @sebavan I just noticed I inadvertently changed this from trilinear to bilinear. Should I change it back?

File: packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts (hunk 1 above)
Comment (bghgary) [reply packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts]: fixed

File: packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts (hunk 2 above)
Comment (bghgary) [reply packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts]: fixed

File: packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts [hunk 4]
```diff
@@ -574,24 +576,18 @@ export class LensRenderingPipeline extends PostProcessRenderPipeline {
...
+        const data = new Uint8Array(size * size * 3);
+        for (let index = 0; index < data.length; ) {
+            const value = Math.floor(Scalar.RandomRange(0.42, 0.58) * 255);
+            data[index++] = value;
+            data[index++] = value;
+            data[index++] = value;
         }
-        (<DynamicTexture>this._grainTexture).update(false);
+
+        const texture = RawTexture.CreateRGBTexture(data, size, size, this._scene, false, false, Constants.TEXTURE_BILINEAR_SAMPLINGMODE);
```
Comment (Popov72) [inline packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts:587]: RGB textures are not supported in WebGPU, you should create a RGBA instead.

File: packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssaoRenderingPipeline.ts (hunk 3 above)
Comment (Popov72) [reply packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssaoRenderingPipeline.ts:351]: No, it does not make any difference, TRI will fallback to BI if there is no mipmaps. In any case, BI is better as we do know there is no mips.

**Review by Popov72 — CHANGES_REQUESTED** [review]

File: packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts (hunk 4 above)
Comment (bghgary) [reply packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/lensRenderingPipeline.ts]: fixed

**Review by Popov72 — APPROVED** [review]

**Pull request merged**